app/assets/javascripts/exercise.ts

```diff
@@ -0,0 +1,477 @@
+/* globals ace */
+import { initTooltips, updateURLParameter } from "util.js";
+import { Toast } from "./toast";
+import GLightbox from "glightbox";
+
+function showLightbox(content): void {
+    const lightbox = new GLightbox(content);
+    lightbox.on("slide_changed", () => {
+        // There might have been math in the image captions, so ask
+        // MathJax to search for new math (but only in the captions).
+        window.MathJax.typeset([".gslide-description"]);
+    });
+    lightbox.open();
+
+    // Transfer focus back to the document body to allow the lightbox to be closed.
+    // https://github.com/dodona-edu/dodona/issues/1759.
+    document.body.focus();
+}
+
+function onFrameMessage(event): void {
+    if (event.message.type === "lightbox") {
+        showLightbox(event.message.content);
+    }
+}
+
+function initLightboxes(): void {
+    let index = 0;
+    const images = [];
+    document.querySelectorAll<HTMLElement>(".activity-description img, a.dodona-lightbox").forEach(el => {
```

Comment:
```suggestion
    document.querySelectorAll<HTMLElement>(".activity-description img, a.dodona-lightbox").forEach((el, index) => {
```

The callback to the [`forEach`](https://developer.mozilla.org/en-US/docs/Web/API/NodeList/forEach) has an optional second argument, which is the index of the current element.